### PR TITLE
Hypothesis testing

### PR DESCRIPTION
**Pull request by wolfram74**
added a type_key to each class and a freshTypeDict method, using that procedurally tested the reversibility of encoding and decoding for each message. If we can reliably transmit the byte string generated by the encoding, it will make an object that produces the same byte string.

### CODE REVIEW

File: src/communications/messages/tests/test_message_factory.py [hunk 1]
```diff
@@ -40,3 +47,34 @@ def testBuildMessageUnsuccessful(self):
...
+    @hypothesis.given(hypothesis.strategies.data())
+    @hypothesis.settings(deadline=None)
+    def testReversibilityOfEncoding(self, data):
+        strat_map = {
+            int:hypothesis.strategies.integers(),
+            str:hypothesis.strategies.characters(),
+            }
+        vals = [item for item in dir(constants) if not item.startswith("__")]
+        for term in vals:
+            if 'SUCCESS' in term:
```
Comment (aaronmorgenegg) [inline src/communications/messages/tests/test_message_factory.py]: If it makes it easier, we could pull all of the MESSAGE_ID constants into it's own file

File: src/communications/messages/tests/test_message_factory.py [hunk 2]
```diff
@@ -40,3 +47,34 @@ def testBuildMessageUnsuccessful(self):
...
+            message_class = MessageFactory.MESSAGE_TYPE_ID_MAP[id_val]
+            types = message_class.freshTypeDict()
+            for argument in types.keys():
+                types[argument] = strat_map[types[argument]]
+            prop = data.draw(
+                hypothesis.strategies.fixed_dictionaries(types)
+                )
+            prop['message_type_id'] = id_val
+            # print(prop)
+            message_instace = MessageFactory.build(**prop)
```
Comment (aaronmorgenegg) [inline src/communications/messages/tests/test_message_factory.py]: typo message_instace -> message_instance

File: src/communications/messages/update_clients.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
 from src.communications.messages.request import Request
 
-
+#hoo boy this is going to get fleshed out
```
Comment (aaronmorgenegg) [inline src/communications/messages/update_clients.py]: Haha... Speaking of, will adding new types like lists or dicts break the hypothesis testing?

**Comment by aaronmorgenegg** [comment]
It'd be nice to pull the encoder/decoder and hypothesis testing into a new directory.

Looks great though, good work

File: src/communications/messages/update_clients.py (hunk 3 above)
Comment (wolfram74) [reply src/communications/messages/update_clients.py]: It'll require updating, but I think I understand it enough now that future upgrades will be faster than setting it up.

**Pull request merged**